File: corpus/fixtures/web-gate/src/limiter.test.ts
// 与被测文件同置的回归（Jest/Vitest 默认约定）。这份文件是 documented_rule_untested
// 的负向对照：它证明回归确实存在，路径启发式必须认得出来。
import { enforce_rate_limit } from "./limiter";

export function test_rate_limit_allows() {
  const r = enforce_rate_limit("u-1", 5);
  if (!r.allowed) throw new Error("limiter failed");
}

File: corpus/fixtures/go-gateway/src/limiter_test.go
package gateway

import "testing"

// Go 的回归只能与被测文件同目录同包——这不是可选风格，是语言约定。
// documented_rule_untested 的负向对照第二域。
func TestEnforceRateLimit(t *testing.T) {
	if !EnforceRateLimit("u-1", 5) {
		t.Fatal("limiter failed")
	}
}
